### DISCUSSION TITLE
Support of HiTechnic Color Sensor V2 in pybricks-micropython

### DISCUSSION THREAD

**Discussion by dianabarrett** [pybricks/support#345] [Q&A] [answered]
Hi!

**EDIT: my sensor is HiTechnic Color Sensor V2**

I'm having problems with using NXT Color Sensor V2 with pybricks.
I've seen different discussions around this, but none of the solutions seem to work for me.

I have an **NXT Color Sensor V2** connected to **S1** of the EV3 brick.
Here is the code I have:
```python
#!/usr/bin/env pybricks-micropython
from pybricks.nxtdevices import ColorSensor
from pybricks.parameters import Port

color_sensor = ColorSensor(Port.S1)
```
When I run this code on my EV3 brick, it just hangs, I have to stop the program manually.
If I put any additional code after the line of `color_sensor` declaration, nothing gets executed.

Also, as you can assume that if I write something like this:
```python
#!/usr/bin/env pybricks-micropython
from pybricks.nxtdevices import ColorSensor
from pybricks.parameters import Port

color_sensor = ColorSensor(Port.S1)
color_sensor._mode('RGB-RAW')
```
The last line of code `color_sensor._mode('RGB-RAW')` is not being executed at all.

**The Question:**
What would be the solution? How can I make the **NXT Color Sensor V2** work within pybricks-micropython?

P.S. I've seen this comment https://github.com/pybricks/support/issues/102#issuecomment-660378293 by @dlech, but I don't understand how I can make this work for my case, that it seems like color_sensor object isn't being created.

**Comment by laurensvalk** [comment, marked as answer]
The NXT color sensor class used in the script above is for the NXT color sensor that comes in the NXT set (https://docs.pybricks.com/en/latest/nxtdevices.html#nxt-color-sensor).

Could it be that you are using the HiTechnic Color Sensor V2 instead? Then you can use this class and its examples:
https://docs.pybricks.com/en/latest/iodevices/ev3devsensor.html

**Reply by dianabarrett** [reply]
Yes!! That was the case, thank you so much!!

**Reply by laurensvalk** [reply]
Awesome! You may need to tweak the examples a bit depending on what you need, but that page and the links provided therein might be enough to get started.

Just let us know if you need any help.
